### PR TITLE
Ports station traits

### PR DESCRIPTION
**Pull request by Archanial**
## About The Pull Request

Ported PRs:
https://github.com/tgstation/tgstation/pull/56309
https://github.com/tgstation/tgstation/pull/57369
https://github.com/tgstation/tgstation/pull/57192
https://github.com/tgstation/tgstation/pull/57224
https://github.com/tgstation/tgstation/pull/57174
https://github.com/tgstation/tgstation/pull/55903
https://github.com/tgstation/tgstation/pull/48760
https://github.com/tgstation/tgstation/pull/57679
https://github.com/tgstation/tgstation/pull/57525
https://github.com/tgstation/tgstation/pull/57532
https://github.com/tgstation/tgstation/pull/58555
https://github.com/tgstation/tgstation/pull/58331
https://github.com/tgstation/tgstation/pull/58681

Adds station traits.
Adds intern announcer.
Minor code improvements and adds support for announcers.

Positive traits

- Lucky winner - Free pizza and beer every 6-12 minutes
- Galactic Grant - Larger starting funds for cargo
- Premium internals boxes - emergency box has flare + radio as bonus
- Bountiful bounties - Bounties pay 20% more
- Strong supply lines - Imports 20% cheaper
- Filled maint - More loot in maint
- Fast shuttle - Cargo shuttle is faster
- Free scarves - Free scarfs if a slot is free
- Deathrattle - Gives everyone (or specific departament) deathrattle implants

Neutral traits

- Bananium shipment - Clown starts with 5 sheets of bananium (Neutral because this helps noone but the clown)
- Unnatural atmosphere - Lava planet can get more restricted gasses
- Ian adventure - Ian teleports to a random spot on the station
- Glitched PDAs - PDA's have a different beep
- Announcer intern - Changes the announcement messages to sound like they're from an intern at Centcom
- Announcer medbot - Changes announcer to medbot

Negative traits

- Carp infestation - Carp event is more common and can start earlier
- Weak supply lines - Imports 20% more expensive
- Blackout - Station lights are partially broken around the station
- Empty maint - Less loot in maint
- Overflow bureacracy mistake - Random overflow job (From a vetted list)
- Late Arrivals - Everyone starts in arrivals
- Random spawns - Random spawn location (by drop pod)
- Slow shuttle - Cargo shuttle is slower
- Bot Language Error - bots speak in random languages
- Hangover - everyone spawns randomly with, after passing out during party

This PR requires #3837 to be merged first.

- [x] port initial PR and some fixes
- [x] port rest of station traits, created later
- [x] wait for pod pr to be merged :pensive: 
- [x] test it
- [x] write description, changelog, credit authors
- [ ] tgui panel for admins to pick traits roundstart

## Why It's Good For The Game

Funny station gimmicks that have random chance to spawn

Special thanks to Qustinnus for making station traits in the first place.

## Changelog 
:cl: Archanial, Qustinnus, MrMelbert. coiax, Rohesie, Ghommie
add: Station trait system, together with 24 traits.
add: Custom announcer support. Medbot and intern announcers.
add: Deathrattle implant. Available for nuke ops for 4tc.
/:cl:

### CODE REVIEW

**Comment by NickEmbased** [comment]
Very epic

**Comment by Mat05usz** [comment]
Very based

**Comment by PowerfulBacon** [comment]
oh nice, was going to do something like this a long time ago but didn't have enough ideas for traits, seems like there is a good number

**Comment by PowerfulBacon** [comment]
also wtf is the intern sound effect stuff

**Comment by Archanial** [comment]
> also wtf is the intern sound effect stuff

Intern announcer was a joke made for 2020 april fools. 
There is station trait that makes the intern replace normal vox announcements. 
I'll add all the info later.

**Comment by PowerfulBacon** [comment]
holy shit anime

**Comment by Archanial** [comment]
The code is more or less ready to review. It's still untested and I'm keeping it draft because I want to make admins be able to force certain traits.

**Comment by Archanial** [comment]
I will finish this I promise

**Comment by Archanial** [comment]
Tested everything works. This is ready for review/testmerge but please don't merge it just yet, I'd like to add some tgui panel for admins to pick station traits.

**Comment by ike709** [comment]
Is this still blocked by TGUI panel?

**Comment by Archanial** [comment]
> 
> 
> Is this still blocked by TGUI panel?

The code is functioning and all of the traits are ported. I can make the panel in some other PR.

**Comment by ike709** [comment]
The heck are all of the map changes for?

**Comment by ike709** [comment]
Doesn't edit corgstation either.

**Comment by ike709** [comment]
> Unique AI - Random lawset at roundstart for AI

Nuke this one, it's already been rejected before.

File: code/controllers/subsystem/processing/station.dm [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	name = "Station"
+	init_order = INIT_ORDER_STATION
+	flags = SS_BACKGROUND
+	runlevels = RUNLEVEL_GAME
+	wait = 5 SECONDS
+
+	///A list of currently active station traits
+	var/list/station_traits = list()
+	///Assoc list of trait type || assoc list of traits with weighted value. Used for picking traits from a specific category.
+	var/list/selectable_traits_by_types = list(STATION_TRAIT_POSITIVE = list(), STATION_TRAIT_NEUTRAL = list(), STATION_TRAIT_NEGATIVE = list())
```
Comment (ike709) [inline code/controllers/subsystem/processing/station.dm]: Set the lists in `Initialize()` below

File: code/datums/announcers/_announcer.dm [hunk 2]
```diff
@@ -0,0 +1,23 @@
+///Data holder for the announcers that can be used in a game, this can be used to have alternative announcements outside of the default e.g.the intern
+/datum/centcom_announcer
+	///Roundshift start audio
+	var/welcome_sounds = list()
```
Comment (ike709) [inline code/datums/announcers/_announcer.dm]: These... aren't lists?

File: code/datums/station_traits/positive_traits.dm [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+	var/scarf_type = pick(scarves)
+
+	living_mob.equip_to_slot_or_del(new scarf_type(living_mob), ITEM_SLOT_NECK)
+
+/datum/station_trait/filled_maint
+	name = "Filled up maintenance"
+	trait_type = STATION_TRAIT_POSITIVE
+	weight = 5
+	show_in_report = TRUE
+	report_message = "Our workers accidentaly forgot more of their personal belongings in the maintenace areas."
```
Comment (ike709) [inline code/datums/station_traits/positive_traits.dm]:
```suggestion
	report_message = "Our workers accidentally forgot more of their personal belongings in the maintenance areas."
```

File: code/game/objects/effects/decals/cleanable/food.dm [hunk 4]
```diff
@@ -35,7 +35,6 @@
 		while(L.loc == src.loc)
 			L.adjustFireLoss(2, TRUE)
 			to_chat(L, "<span class='danger'>The salt! It burns!</span>")
-			sleep(20)
```
Comment (ike709) [inline code/game/objects/effects/decals/cleanable/food.dm]: ?

**Comment by ike709** [comment]
This adds multiple MB worth of soundfiles. Maybe trim the number of options a bit?

**Comment by Archanial** [comment]
> 
> 
> The heck are all of the map changes for?

For hangover trait. I made this PR long before corg was merged.



> 
> 
> This adds multiple MB worth of soundfiles. Maybe trim the number of options a bit?

Is their size really a problem? It adds 2 new announcers and moves old one.

edit: this adds exactly 7,61 MB of sound files.

**Comment by ike709** [comment]
PR fully reviewed. Expect lots of testmerging after reviews are addressed.

**Comment by ike709** [comment]
You need to update corgstation.

**Comment by ike709** [comment]
> Is their size really a problem? It adds 2 new announcers and moves old one.

Upon doing some basic math it's only like 3 MB so I *guess* it's fine.

File: code/controllers/subsystem/processing/station.dm [hunk 5]
```diff
@@ -8,12 +8,14 @@ PROCESSING_SUBSYSTEM_DEF(station)
 	///A list of currently active station traits
 	var/list/station_traits = list()
 	///Assoc list of trait type || assoc list of traits with weighted value. Used for picking traits from a specific category.
-	var/list/selectable_traits_by_types = list(STATION_TRAIT_POSITIVE = list(), STATION_TRAIT_NEUTRAL = list(), STATION_TRAIT_NEGATIVE = list())
+	var/list/selectable_traits_by_types = list()
```
Comment (ike709) [inline code/controllers/subsystem/processing/station.dm]:
```suggestion
	var/list/selectable_traits_by_types
```

File: code/controllers/subsystem/processing/station.dm [hunk 6]
```diff
@@ -8,12 +8,14 @@ PROCESSING_SUBSYSTEM_DEF(station)
 	///A list of currently active station traits
 	var/list/station_traits = list()
```
Comment (ike709) [inline code/controllers/subsystem/processing/station.dm]:
```suggestion
	var/list/station_traits
```

File: code/controllers/subsystem/processing/station.dm [hunk 7]
```diff
@@ -8,12 +8,14 @@ PROCESSING_SUBSYSTEM_DEF(station)
...
 	///Assoc list of trait type || assoc list of traits with weighted value. Used for picking traits from a specific category.
-	var/list/selectable_traits_by_types = list(STATION_TRAIT_POSITIVE = list(), STATION_TRAIT_NEUTRAL = list(), STATION_TRAIT_NEGATIVE = list())
+	var/list/selectable_traits_by_types = list()
 	///Currently active announcer. Starts as a type but gets initialized after traits are selected
 	var/datum/centcom_announcer/announcer = /datum/centcom_announcer/default
 
 /datum/controller/subsystem/processing/station/Initialize(timeofday)
 
+	selectable_traits_by_types = list(STATION_TRAIT_POSITIVE = list(), STATION_TRAIT_NEUTRAL = list(), STATION_TRAIT_NEGATIVE = list())
```
Comment (ike709) [inline code/controllers/subsystem/processing/station.dm:18]:
```suggestion
	station_traits = list()
	selectable_traits_by_types = list(STATION_TRAIT_POSITIVE = list(), STATION_TRAIT_NEUTRAL = list(), STATION_TRAIT_NEGATIVE = list())
```

File: code/datums/announcers/_announcer.dm [hunk 8]
```diff
@@ -1,11 +1,11 @@
...
-	var/welcome_sounds = list()
+	var/list/welcome_sounds = list()
 	///Sounds made when announcement is receivedc
-	var/alert_sounds = list()
+	var/list/alert_sounds = list()
 	///Sounds made when command report is received
-	var/command_report_sounds = list()
+	var/list/command_report_sounds = list()
 	///Event audio, can be used for specific event announcements and is assoc key - sound. If no sound is found the default is used.area
 	var/event_sounds = list()
```
Comment (ike709) [inline code/datums/announcers/_announcer.dm]: Missed one.

Comment (Archanial) [inline code/datums/announcers/_announcer.dm]:
```suggestion
	var/list/event_sounds = list()
```

**Comment by Archanial** [comment]
Random AI lawset trait has been removed.

File: code/modules/events/meteor_wave.dm [hunk 9]
```diff
@@ -47,7 +47,7 @@
 			kill()
 
 /datum/round_event/meteor_wave/announce(fake)
-	priority_announce("Meteors have been detected on collision course with the station. Estimated time until impact: 10 MINUTES. Anti-meteor point defense is available for purchase via the station's cargo shuttle.", "Meteor Alert", 'sound/ai/meteors.ogg')
+	priority_announce("Meaty ores have been detected on collision course with the station.", "Oh crap, get the mop.", ANNOUNCER_METEORS)
```
Comment (ike709) [inline code/modules/events/meteor_wave.dm]: The string shouldn't be changed, just that last arg.

**Comment by ike709** [comment]
image

>announcer_meteors

**Comment by ike709** [comment]
This is good to go aside from whatever new fix PRs you need to add.

**Pull request merged**